test/integration/test_integration_basics.py

```diff
@@ -441,3 +442,71 @@ def test_bad_arg_error_messages(http_test_server_fixture):
       expect_failure=True,
       as_json=False)
   assert "Bad argument: Termination predicate 'a:a' has an out of range threshold." in err
+
+
+def test_multiple_backends_http_h1(multi_http_test_server_fixture):
+  """
+  Runs the CLI configured to use plain HTTP/1 against multiple test servers, and sanity
+  checks statistics from both client and server.
+  """
+  nighthawk_client_args = [
+      "--multi-target-path", "/", "--duration", "100", "--termination-predicate",
+      "benchmark.http_2xx:24"
+  ]
+  for uri in multi_http_test_server_fixture.getAllTestServerRootUris():
+    nighthawk_client_args.append("--multi-target-endpoint")
+    nighthawk_client_args.append(uri.replace("http://", "").replace("/", ""))
+
+  parsed_json, stderr = multi_http_test_server_fixture.runNighthawkClient(nighthawk_client_args)
+
+  counters = multi_http_test_server_fixture.getNighthawkCounterMapFromJson(parsed_json)
+  assertCounterEqual(counters, "benchmark.http_2xx", 25)
```

Comment:
This is a fairly large amount of assertions. Are all of them related to verifying the multi-backend functionality or are we combining multiple behaviors in this test?

Can you help me understand which of these assertions are actually verifying the multi-backend behavior?

Reply:
The assertions involving 3 connections and the total number of responses are relevant to making sure multi-target is working correctly. I relaxed the ones about the total bytes, but we might still want check if there are more than zero bytes as a sanity check. Removed unneeded assertions about the number of stats.